### PR TITLE
fix(ctl): make `./risedev ctl profile` work on linux again

### PR DESCRIPTION
**Pull request by yuhao-su**
I hereby agree to the terms of the [Singularity Data, Inc. Contributor License Agreement](https://gist.github.com/skyzh/0663682a70b0edde7ae991492f2314cb#file-s9y_cla).

## What's changed and what's your intention?

as title

## Checklist

- [x] I have written necessary rustdoc comments
- [x] I have added necessary unit tests and integration tests
- [x] All checks passed in `./risedev check` (or alias, `./risedev c`)

## Documentation

use  `RW_META_ADDR=xxx ./risedev ctl profile --sleep [time]` to profile all compute nodes.

It might not work if the CNs consumed all resources and failed to respond to profiling RPCs.

## Refer to a related PR or issue link (optional)
#4765

### CODE REVIEW

**Review by fuyufjh — APPROVED** [review]

File: src/ctl/src/cmd_impl/profile.rs [hunk 1]
```diff
@@ -39,29 +41,40 @@ pub async fn profile(sleep_s: u64) -> anyhow::Result<()> {
...
+                    tracing::error! {"Failed to get profiling result from {} with error {}", node_name, err.to_string()};
+                }
             }
-        }
+            Ok::<_, RwError>(())
+        };
+        profile_futs.push(fut);
     }
 
+    let _ = join_all(profile_futs).await;
```
Comment (BugenZhao) [inline src/ctl/src/cmd_impl/profile.rs:76]:
```suggestion
    join_all(profile_futs).await?;
```

File: src/ctl/src/cmd_impl/profile.rs [hunk 2]
```diff
@@ -39,29 +41,40 @@ pub async fn profile(sleep_s: u64) -> anyhow::Result<()> {
...
+            let node_name = format!(
+                "compute-node-{}-{}",
+                host_addr.get_host().replace('.', "-"),
+                host_addr.get_port()
+            );
+            let svg_file_name = format!("{}.svg", node_name);
+            match response {
+                Ok(ProfilingResponse { result }) => {
+                    let mut file = File::create(dir_path_ref.join(svg_file_name))?;
+                    file.write_all(&result)?;
```
Comment (BugenZhao) [inline src/ctl/src/cmd_impl/profile.rs:65]: Should we use `tokio::fs` here? 🤣 Not a big deal.

**Review by BugenZhao — COMMENTED** [review]
LGTM

**Comment by yuhao-su** [comment]
> LGTM.
> 
> The diff in `profile.rs` seems to parallelize it, right? So the primary fix is from upstream of `pprof`?

Yes. I just found the parallelism problem when testing and fix it.

**Pull request merged**